cpp/cmake/modules/raft_export.cmake

```diff
@@ -202,7 +203,8 @@ function(raft_export type project_name)
     endif()
 
     # Install everything we have generated
-    install(DIRECTORY "${scratch_dir}/" DESTINATION "${install_location}")
+    install(DIRECTORY "${scratch_dir}/" DESTINATION "${install_location}"
```

Comment:
This is wrong as it will capture the dependencies files like `raft-distance-dependencies.cmake` in the `raft` component and not the `raft_distance` component.

I think we want to change the `rapids_export_write_dependencies` in the root CMakeLists.txt to write to a per component folder, and update `raft_export` to map each component name to a install call of the associated component.